### PR TITLE
Extracted Model content from StyledTextArea (view) into its own model class (StyledTextAreaModel)

### PR DESCRIPTION
**Pull request by JordanMartinez**
So, I got to it sooner. (It's also has a somewhat cleaner commit history).

### CODE REVIEW

File: richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java [hunk 1]
```diff
@@ -137,31 +137,6 @@
...
-     *                *
-     ******************/
-
-    /**
-     * Indicates whether this text area can be edited by the user.
-     * Note that this property doesn't affect editing through the API.
-     */
-    boolean isEditable();
-    void setEditable(boolean value);
-    BooleanProperty editableProperty();
```
Comment (TomasMikula) [inline richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java]: We should preserve the Javadoc comment by moving it to StyledTextArea.

File: richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java [hunk 2]
```diff
@@ -137,31 +137,6 @@
...
-
-    /**
-     * When a run of text exceeds the width of the text region,
-     * then this property indicates whether the text should wrap
-     * onto another line.
-     */
-    boolean isWrapText();
-    void setWrapText(boolean value);
-    BooleanProperty wrapTextProperty();
```
Comment (TomasMikula) [inline richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java]: We should preserve the Javadoc comment by moving it to StyledTextArea.

File: richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java (hunk 1 above)
Comment (JordanMartinez) [inline richtextfx/src/main/java/org/fxmisc/richtext/TextEditingArea.java]: Oops. Good point!

**Comment by TomasMikula** [comment]
Looking good :+1: Thanks!

**Comment by JordanMartinez** [comment]
☺

**Pull request merged**